Vagrantfile

```diff
@@ -0,0 +1,115 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+  config.vm.box = "boxcutter/ubuntu1604"
+
+
+    config.vm.network "forwarded_port", guest: 3000, host: 3000, auto_correct: true
+    config.vm.network "forwarded_port", guest: 80, host: 8080, auto_correct: true
+    config.vm.synced_folder ".", "/vagrant", disabled: true
+    config.vm.synced_folder ".", "/home/vagrant/app", type: "rsync", rsync__exclude: ".git/"
+
+    config.vm.provider "virtualbox" do |vb|
+     # Display the VirtualBox GUI when booting the machine
+     # vb.gui = true
+
+     # Customize the amount of memory on the VM:
+     vb.memory = "1024"
+    end
+
+
+    config.vm.provision "shell", privileged: true, inline: <<-SHELL
+      apt-get -y update
+      apt-get -y install build-essential
+      apt-get -y install zlib1g-dev git-core nodejs
+
+      # Install Apache and MySQL
+      apt-get install -y apache2
+      debconf-set-selections <<< 'mysql-server-5.5 mysql-server/root_password password root'
+      debconf-set-selections <<< 'mysql-server-5.5 mysql-server/root_password_again password root'
+      apt-get -y install mysql-server mysql-client libmysqlclient-dev
+
+      apt-get install -y autoconf bison build-essential libssl-dev libyaml-dev
+      apt-get install -y libreadline6-dev zlib1g-dev libncurses5-dev
+      apt-get install -y libgdbm3 libgdbm-dev libsqlite3-dev
+      apt-get install -y libreadline-dev libssl-dev libffi-dev libcurl3
+      apt-get install -y libxml2-dev libxslt1-dev python-software-properties
+
+      echo "++++++++++ IPTABLES ++++++++++++++"
```

Comment:
Do we need `iptables` set up? I think the default vagrant network uses NAT which should only be accessible from the host anyways.